packages/sso/src/saml/assertions.ts

```diff
@@ -0,0 +1,61 @@
+import { APIError } from "better-auth/api";
+import { countAllNodes, xmlParser } from "./parser";
+
+export interface AssertionCounts {
+	assertions: number;
+	encryptedAssertions: number;
+	total: number;
+}
+
+/** @lintignore used in tests */
+export function countAssertions(xml: string): AssertionCounts {
+	let parsed: unknown;
+	try {
+		parsed = xmlParser.parse(xml);
+	} catch {
+		throw new APIError("BAD_REQUEST", {
+			message: "Failed to parse SAML response XML",
+			code: "SAML_INVALID_XML",
+		});
+	}
+
+	const assertions = countAllNodes(parsed, "Assertion");
+	const encryptedAssertions = countAllNodes(parsed, "EncryptedAssertion");
+
+	return {
+		assertions,
+		encryptedAssertions,
+		total: assertions + encryptedAssertions,
+	};
+}
+
+export function validateSingleAssertion(samlResponse: string): void {
+	let xml: string;
+	try {
+		xml = Buffer.from(samlResponse, "base64").toString("utf-8");
```

Comment:
no buffer because it's Node.js only feature, Please use the class from our utils

Reply:
Ah its great that we alredy have a class with this. Missed that, ty